packages/nextjs/hooks/noir/useProofGenerator.ts

```diff
@@ -29,41 +29,34 @@ function getPublicInputsLength(parameters: CircuitAbiParameters) {
   return parameters
     .filter(param => param.visibility === "public")
     .reduce((acc, param) => {
-      return acc + (param.type.length || 1);
+      if (param.type.kind === "array") {
+        return acc + param.type.length;
+      } else {
+        return acc + 1;
+      }
     }, 0);
 }
 
 export const generateProof = async (circuitName: CircuitName, parsedArgs: ParsedArgs) => {
   isGeneratingProof = true;
-  console.log("🧠 start");
-  console.log("parsedArgs", parsedArgs);
   const noir = new NoirBrowser();
   try {
-    const circuit = circuits[circuitName];
+    const circuit = circuits[circuitName] as Circuit<CircuitName>;
```

Comment:
is it needed to cast here?

Reply:
Well, it was one way to get the type-checking to pass. The types are not set-up perfectly yet, but I want to be able to commit at least. I'm open to another solution!

Reply:
Good to leave a TODO at lest in this PR

Reply:
https://github.com/Kryha/scaffold-eth-2-noir/pull/4/commits/cd47494fa9c7fc44e348deb77933e72b53f0faff